app/models/champs/yes_no_champ.rb

```diff
@@ -20,36 +20,10 @@
 #  parent_id                      :bigint
 #  type_de_champ_id               :integer
 #
-class Champs::YesNoChamp < Champ
-  def search_terms
-    if true?
-      [libelle]
-    end
-  end
-
-  def to_s
-    processed_value
-  end
-
-  def for_tag
-    processed_value
-  end
-
-  def for_export
-    processed_value
-  end
+class Champs::YesNoChamp < Champs::BooleanChamp
+  validates :value, inclusion: ["true", "false"], allow_nil: true, allow_blank: false
```

Comment:
on a aussi de la bad data ici, il faudrait faire un tache de nettoyage
blank -> nil
string not [true, false] -> false

Reply:
C'est aussi dans https://github.com/demarches-simplifiees/demarches-simplifiees.fr/pull/8320